src/User/Infra/Security/SecurityUserProvider.php

```diff
@@ -16,7 +16,7 @@
 /**
  * @author Roland Franssen <franssen.roland@gmail.com>
  */
-final class SecurityUserProvider implements UserProviderInterface
+class SecurityUserProvider implements UserProviderInterface
```

Comment:
Do we really want to make this extendable?

Reply:
Dont think so :) a [decorator](https://en.wikipedia.org/wiki/Decorator_pattern#PHP would be nice.

@DevMDamien does this work out?

```php
namespace MsgPhp\User\Infra\Security\Jwt;

use MsgPhp\User\Infra\Security\Jwt\SecurityUserProvider as BaseSecurityUserProvider;

class MsgPhp\User\Infra\Security\Jwt\SecurityUserProvider
implements UserProviderWithPayloadSupportsInterface
{
   private $provider;
   public function __construct(BaseSecurityUserProvider $provider) {
      $this->provider = $provider;
  }
}
```

and go from there (all default methods can be forwarded to the decorated provider).

Reply:
Fair enough. I just trigger nervously every time I see a `-final` in a diff :wink: 

Mainly due to having had to reign in out of control APIs before, but I don't see that as problematic here. :+1:

Reply:
I'm ok with that, I will fix it during the day.